src/reports/package/reporter-factory.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import axe from 'axe-core';
+import { getCardViewData } from 'common/rule-based-view-model-provider';
+import { generateUID } from 'common/uid-generator';
+import { convertScanResultsToUnifiedResults } from 'injected/adapters/scan-results-to-unified-results';
+import { convertScanResultsToUnifiedRules } from 'injected/adapters/scan-results-to-unified-rules';
+import { AutomatedChecksReportSectionFactory } from 'reports/components/report-sections/automated-checks-report-section-factory';
+import { getDefaultAddListenerForCollapsibleSection } from 'reports/components/report-sections/collapsible-script-provider';
+import { AxeResultReport, AxeResultReportDeps } from 'reports/package/axe-results-report';
+import { Reporter } from 'reports/package/reporter';
+import { ReactStaticRenderer } from 'reports/react-static-renderer';
+import { ReportHtmlGenerator } from 'reports/report-html-generator';
+import { CheckMessageTransformer } from 'scanner/check-message-transformer';
+import { configuration } from 'scanner/custom-rule-configurations';
+import { DocumentUtils } from 'scanner/document-utils';
+import { HelpUrlGetter } from 'scanner/help-url-getter';
+import { MessageDecorator } from 'scanner/message-decorator';
+import { ResultDecorator } from 'scanner/result-decorator';
+import { getPropertyConfiguration } from '../../common/configs/unified-result-property-configurations';
+import { DateProvider } from '../../common/date-provider';
+import { EnvironmentInfoProvider } from '../../common/environment-info-provider';
+import { initializeFabricIcons } from '../../common/fabric-icons';
+import { GetGuidanceTagsFromGuidanceLinks } from '../../common/get-guidance-tags-from-guidance-links';
+import { FixInstructionProcessor } from '../../injected/fix-instruction-processor';
+import AccessibilityInsightsReport from './accessibilityInsightsReport';
+
+const axeResultsReportGenerator = (results: axe.AxeResults, options: AccessibilityInsightsReport.ReportOptions) => {
+    const { browserVersion, browserSpec, pageTitle: targetPageTitle } = options;
+    const axeVersion = results.testEngine.version;
+
+    initializeFabricIcons();
```

Comment:
Is this the right place to call `initializeFabricIcons`? It seems to me that this call should only be needed when you actually generate the html report (basically `src/reports/package/axe-results-report.ts`). Since this call have side effects, maybe it's a good idea to move where said side effects are actually needed, right?

Reply:
Yeah. That's another head scratcher for me. You get a warning when you call it more than once to generate the report.
 I hate this global state kind of thing.

Reply:
I disabled the warnings so that is no longer an issue.